### PR TITLE
Add bindings for aes-gcm encrypt/decrypt

### PR DESCRIPTION
**Pull request by olorin**
! @markhibberd @erikd-ambiata
/jury approve @markhibberd @ambiata-ci

### CODE REVIEW

File: test/Test/IO/Tinfoil/AEAD/AESGCM/Foreign.hs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    Left err -> failWith . T.pack $ show err
+
+prop_aes256Gcm_encrypt :: Cleartext -> AssociatedData -> SymmetricKey -> GcmIv -> Property
+prop_aes256Gcm_encrypt clear ad sk iv =
+  testSodium $ \m ->
+    let
+      ct = aes256GcmEncrypt m clear ad sk iv
+    in
+    (BS.isInfixOf (unCleartext clear) (unAuthenticatedCiphertext ct)) === False
+
```
Comment (erikd-ambiata) [inline test/Test/IO/Tinfoil/AEAD/AESGCM/Foreign.hs]: Tiny minor bikeshedding thing. Pretty sure, the outer parens aren't needed there. Just thought I'd mention it. Fixing it is totally up to you.

**Comment by erikd-ambiata** [comment]
:+1: on :green_apple:

File: test/Test/IO/Tinfoil/AEAD/AESGCM/Foreign.hs (hunk 1 above)
Comment (olorin) [inline test/Test/IO/Tinfoil/AEAD/AESGCM/Foreign.hs]: Will fix, thanks.

**Comment by markhibberd** [comment]
👍

**Pull request merged**